Libplanet/PoS/Control/Bond.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using Libplanet.Action;
+using Libplanet.Assets;
+
+namespace Libplanet.PoS
+{
+    internal static class Bond
+    {
+        internal static (IAccountStateDelta, FungibleAssetValue) Execute(
+            IAccountStateDelta states,
+            FungibleAssetValue consensusToken,
+            Address validatorAddress,
+            Address delegationAddress)
+        {
+            // TODO: Failure condition
+            // 1. Validator does not exist
+            // 2. Exchange rate is invalid(validator has no tokens but there are outstanding shares)
+            // 3. Amount is less than the minimum amount
+            // 4. Delegator does not have sufficient consensus token (fail or apply maximum)
+            if (!consensusToken.Currency.Equals(Asset.ConsensusToken))
+            {
+                throw new ArgumentException();
+            }
+
+            Validator validator = ValidatorCtrl.GetValidatorByAddr(states, validatorAddress);
+
+            // If validator share is zero, exchange rate is 1
+            // Else, exchange rate is validator share / token
+            FungibleAssetValue issuedShare
+                = ValidatorCtrl.ShareFromConsensusToken(states, validator.Address, consensusToken)
+                ?? throw new InvalidOperationException();
```

Comment:
Plus, IMO this also needs a bit more descriptive...

Reply:
Applied on commit 8045f68c000f18f2e9b8608ae7c2b33ccfc96cd0.